flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/stream/table/DataGeneratorConnectorITCase.java

```diff
@@ -71,4 +71,19 @@ public void testTypes() throws Exception {
 
         Assert.assertEquals("Unexpected number of results", 10, results.size());
     }
+
+    @Test
+    public void testLimit() throws Exception {
+        tEnv().executeSql(TABLE);
+
+        List<Row> results = new ArrayList<>();
+
+        try (CloseableIterator<Row> iter = tEnv().executeSql("select * from datagen_t limit 5").collect()) {
```

Comment:
Since this table is still defined with `number-of-rows = 5`, this test doesn't actually assert that the limit keyword does anything.

Reply:
This table actually defined with `'number-of-rows' = '10'`.

Reply:
Agree with @Airblader, please add a test with a new table definition that does not include `number-of-rows`, as it also tests that the validation phase doesn't fail because of missing `number-of-rows`

Reply:
OK